In [1]:
import pandas as pd
df=pd.read_csv("usarrests.csv")

df.describe()

,Murder,Assault,UrbanPop
count,50.00000,49.000000,50.00000
mean,7.78800,182.183673,74.20000
std,4.35551,130.877435,73.40828
min,0.80000,45.000000,6.00000
25%,4.07500,109.000000,53.25000
50%,7.25000,159.000000,66.00000
75%,11.25000,249.000000,77.75000
max,17.40000,879.000000,570.00000


In [2]:
df.isna().sum()

Unnamed: 0    0
Murder        0
Assault       1
UrbanPop      0
dtype: int64

In [3]:
df[df["UrbanPop"]>100]

,Unnamed: 0,Murder,Assault,UrbanPop
14,Iowa,2.2,56.0,570


In [4]:
df[df["Assault"].isna()]

,Unnamed: 0,Murder,Assault,UrbanPop
9,Georgia,17.4,NaN,60


The urban population can not exceed 100 percent which is why Iowa is suspect. It could possibly be that whoever entered the data accidentally added a 0, but unless otherwise proven, this is an unreliable number. 

Georgia has no data for Assault despite having it for Murder and Urban Population. This data is unreliable, but can still be possible.

South Carolina has 879 Assaults out of 10,000 people which an extreme outlier for Assault. It would make the Assault per 10,000 people higher than it actually seems, which would still be possible.

New York has an urban population of 6 which is another extreme outlier, the data suggests the average would be 74 and have a median of 66 so we would expect most of the distribution to be around that range. This is also significantly lower than the next lowest which is Vermont at an urban population of 32. The data is highly suspect and would likely be impossible.

The last suspicious value is North Dakota with 0.8 for Murder and 45 for Assault having 44 for Urban Population. This is extreme but still possible.

Repo Link: https://github.com/KyeSPu/cs82a-portfolio